sled-agent/src/sled_agent.rs

```diff
@@ -424,14 +444,34 @@ impl SledAgent {
     }
 
     async fn ensure_scrimlet_services_active(&self, log: &Logger) {
-        // TODO(https://github.com/oxidecomputer/omicron/issues/823): Launch the switch zone, with
-        // Dendrite, MGS, and any other services we want to enable.
-        warn!(log, "Activating scrimlet services not yet implemented");
+        info!(log, "Ensuring scrimlet services (enabling services)");
+
+        let services = match self.inner.config.stub_scrimlet {
+            Some(_) => {
+                vec![ServiceType::Dendrite { asic: DendriteAsic::TofinoStub }]
+            }
+            None => {
+                vec![
+                    ServiceType::Dendrite { asic: DendriteAsic::TofinoAsic },
+                    ServiceType::Tfport { pkt_source: "tfpkt0".to_string() },
```

Comment:
I'm not crazy about hardcoding `tfpkt0` here.  That's almost certainly going to be different for softnpu.  It will also be different if I ever get this running with the tofino simulator (which, tbh, probably isn't worth the effort.).  I guess we could put it in package-manifest.toml, but that seems weird.

I don't have any better options to suggest, so maybe hardcoding is the way to go, for now at least.

Reply:
I'm split on this - this is the location where we have made the decision to launch the switch zone, but the question of "what flavor of switch zone should we launch" is currently being inferred from the sled agent's configuration file. 

This does need to be a piece of information that's self-contained within the sled agent, because of the autonomous nature of launching the switch zone, so either "hardcoded" or "in a config file" seem like the only decent options to me.

My inclination was:
- Provide a mechanism in the config file to toggle between flavor of dendrite (ASIC, stub, softnpu)
- Automatically load the right "profile" for those flavors internally, where we can. I thought `tfpkt0` was one of those parameters.

Reply:
FWIW, that analysis seems sound @smklein

Reply:
Works for me.  If we ever need to make this more flexible, we can figure out the right mechanism then.